packages/baseclient-components/src/container/LayerLegendAccordion/LayerLegendAccordion.tsx

```diff
@@ -0,0 +1,496 @@
+        {legends}
+      </div>
+    );
+  }
+
+  /**
+   * Returns css class name for visibility class of all layers
+   *
+   * @param {OlLayer[]} The OpenLayes layers to check
+   * @param {string} The css class name
+   */
+  getLayerVisiblilityClassName(layers: any[]) {
+    if (!layers) {
+      return 'fa fa-eye-slash all-layers-handle';
+    }
+    const filteredLayers = layers.filter(this.props.treeNodeFilter!);
+    if (!filteredLayers) {
+      return 'fa fa-eye-slash all-layers-handle';
+    }
+    const numLayers = filteredLayers.length;
+    let visibleLayers = 0;
+    filteredLayers.forEach(l => {
+      if (l.getVisible() === true) {
+        visibleLayers++;
+      }
+    });
+    if (visibleLayers === 0) {
+      return 'fa fa-eye-slash all-layers-handle';
+    }
+
+    return visibleLayers === numLayers ? 'fa fa-eye all-layers-handle' : 'fa fa-eye some-layers all-layers-handle';
+  }
+
+  /**
+   * Handler called when visiblity of all layers is changed
+   *
+   * @param {OlLayer[]} The OpenLayes layers to change visiblity for
+   */
+  onAllLayersVisibleChange(mapLayers: any[]) {
+    const filteredLayers = mapLayers.filter(this.props.treeNodeFilter!);
+    const layerVisibilityClassName: string = this.getLayerVisiblilityClassName(mapLayers);
+    let visibility = false;
+    if (layerVisibilityClassName.indexOf('some-layers') > -1 ) {
+      visibility = true;
+    }
+    if (layerVisibilityClassName.indexOf('fa-eye-slash') > -1) {
+      visibility = true;
+    }
+    // update all layers
+    filteredLayers.forEach(l => l.setVisible(visibility));
+
+    // we need to do this since legend needs to be redrawn
+    this.forceUpdate();
+  }
+
+  /**
+   * Handler called if visibility of a certain layers is changed
+   * @param {OlLayer[]} layer The layer for which the visibility has to be changed
+   */
+  onLayerTreeNodeVisibilityChange(layer: any) {
+    const nextLayerVisibility = !layer.getVisible();
+    layer.setVisible(nextLayerVisibility);
+
+    // we need to do this since legend needs to be redrawn
+    this.forceUpdate();
+  }
+
+  /**
+   * Change handler for accordion
+   * @param {string[]} activeKeys The keys of the panels that should be visible afterwards
+   */
+  onAccordionChange(activeKeys: string[]) {
+    this.setState({ activeKeys });
+  }
+
+  /**
+   * The render function
+   */
+  render() {
+    const {
+      map,
+      t
+    } = this.props;
+
+    const {
+      activeKeys,
+      isCollapsed
+    } = this.state;
+
+    const mapLayers = this.props.map.getLayers().getArray();
+    const layerVisibilityClassName = this.getLayerVisiblilityClassName(mapLayers);
+
+    return (
+        isCollapsed ? <SimpleButton
+          icon="list"
+          shape="circle"
+          tooltip={t('LayerLegendAccordion.collapseAccordionTooltipText')}
+          onClick={() => this.setState({ isCollapsed: false })}
+          tooltipPlacement={'bottom'}
+          className="collapse-icon"
+        /> :
```

Comment:
Having the `SimpleButton` inside this component makes it hard to reuse (in cases where you want to trigger the visibility with a custom behaviour). I would suggest to move this button to an own component and to handle the `isCollapsed` prop via the global state. This way you could also position the button relative to the map to fix the `TODO` in [here](https://github.com/terrestris/react-geo-baseclient/pull/373/files#diff-861f15ed92dde87e099c66a69a5db2deR3).

Reply:
:heavy_check_mark: , introduced `LayerLegendAccordionToggleButton` in 913024be